src/js/common/utils/lodash-replacements/set.js

```diff
@@ -0,0 +1,70 @@
+import deconstructPath from './deconstructPath';
+import cloneDeep from './cloneDeep';
+
+/**
+ * Same as `set`, but uses the level param to determine when to clone and give a more helpful error message.
+ *
+ * @param {Array|Obect} object
+ * @param {Array|string} path
+ * @param {*} value
+ * @param {Number} level  How many times we've recursed
+ * @return {Object} A new object with the appropriate value set
+ */
+function baseSet(path, value, object, level = 0) {
+  let arrayPath = Array.isArray(path) ? path : deconstructPath(path);
```

Comment:
I'm curious why this is done at each level vs once at the beginning.

Reply:
Because I added the `level` bit later and forgot about this. I'll fix it. 👍